packages/cdk/lib/transcription-service.ts

```diff
@@ -576,5 +583,86 @@ export class TranscriptionService extends GuStack {
+				default: `/${props.stage}/investigations/alarmTopicArn`,
+			},
+		).valueAsString;
+		if (isProd) {
+			const alarms = [
+				// alarm when a message is added to the dead letter queue
+				// note that queue metrics go to 'sleep' if it is empty for more than 6 hours, so it may take up to 16 minutes
+				// for this alarm to trigger - see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-monitoring-using-cloudwatch.html
+				new Alarm(this, 'DeadLetterQueueAlarm', {
+					alarmName: `transcription-service-dead-letter-queue-${props.stage}`,
+					metric:
+						transcriptionDeadLetterQueue.metricApproximateNumberOfMessagesVisible(
+							{ period: Duration.minutes(1), statistic: 'max' },
+						),
+					threshold: 1,
+					evaluationPeriods: 1,
+					actionsEnabled: true,
+					alarmDescription: `A transcription job has been sent to the dead letter queue. This may be because ffmpeg can't convert the file (maybe it's a JPEG) or because the transcription job has failed multiple times.`,
+					treatMissingData: TreatMissingData.IGNORE,
+					comparisonOperator:
+						ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+				}),
+				// alarm when failure metric is greater than 0
+				new Alarm(this, 'FailureAlarm', {
+					alarmName: `transcription-service-failure-${props.stage}`,
+					//  reference the custom metric created in metrics.ts library
+					metric: new Metric({
+						namespace: 'TranscriptionService',
+						metricName: 'Failure',
+						dimensionsMap: {
+							Stage: props.stage,
+						},
+						statistic: 'sum',
+						period: Duration.minutes(1),
+					}),
+					threshold: 1,
+					comparisonOperator:
+						ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+					evaluationPeriods: 1,
+					actionsEnabled: true,
+					alarmDescription: 'A transcription service failure has occurred',
+					treatMissingData: TreatMissingData.IGNORE,
+				}),
+				// alarm when instances have been running in the worker asg for more than 5 hours
+				new Alarm(this, 'WorkerInstanceAlarm', {
+					alarmName: `transcription-service-worker-instances-${props.stage}`,
+					// this doesn't actually create the metric - just a reference to it
+					metric: new Metric({
+						namespace: 'AWS/AutoScaling',
+						metricName: 'GroupTotalInstances',
+						dimensionsMap: {
+							AutoScalingGroupName: transcriptionWorkerASG.autoScalingGroupName,
+						},
+						statistic: 'min',
+						period: Duration.minutes(5),
+					}),
+					threshold: 1,
+					comparisonOperator:
+						ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+					evaluationPeriods: 5 * 12, // 5 hours as metric has period of 5 minutes
+					actionsEnabled: true,
+					alarmDescription:
+						'There has been more than 1 worker instance running for 5 hours - this will have significant cost implications. Please check that all running workers are doing something useful.',
+					treatMissingData: TreatMissingData.IGNORE,
+				}),
+			];
```

Comment:
I'm trying to get my head around this last alarm. From what I understand from this alarm, during 5 hours, if there are different instances that are being scaled out (not  only 1 instance running for 5 hours), this alarm would still be triggered right?

Reply:
Yes - every 5 minutes we check the minimum number of instances that were running during that 5 minute period, if it is >0 for every 5 minute period over 5 hours then the alarm triggers.

I did this because it's a lot easier than tracking whether one specific instance has been running for more than 5 hours. I still think this is a pretty unlikely situation - because the alarm will reset as soon as we have 0 workers for 60 seconds.

Reply:
Thanks @philmcmahon for explaining, it makes sense 👍